app/models/spree/variant_decorator.rb

```diff
@@ -85,6 +85,16 @@ def self.indexed
     ]
   end
 
+  def self.active(currency = nil)
+    # "where(id:" is necessary so that the returned relation has no includes
+    # The relation without includes will not be readonly and allow updates on it
+    where(id: joins(:prices).
```

Comment:
I just remembered another workaround for Rails 3:

```suggestion
    where("spree_variants.id in (?)", joins(:prices).
```

Reply:
nice!!!!
I confirmed this works:
```ruby
puts Spree::Variant.where("spree_variants.id in (?)", [1]).where("spree_variants.id in (?)": [1,2,3]).to_sql
SELECT "spree_variants".* FROM "spree_variants"  WHERE "spree_variants"."id in (?)" IN (1, 2, 3) AND ("spree_variants".deleted_at IS NULL) AND (spree_variants.id in (1))
 => nil 

```